documentation/src/main/asciidoc/reference/search-dsl-highlighting.asciidoc

```diff
@@ -0,0 +1,416 @@
+.Configuring fields for highlighting
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/highlighting/Book.java[tags=basics]
+----
+<1> Allow plain and unified highlighter types to be applied to the `title` field.
+<2> Allow any highlighter types to be applied to the `description` field.
+====
+
+[[search-dsl-highlighting-using-highlight-projection]]
+.Using highlight projection
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/highlighting/HighlightingDslIT.java[tags=basics]
+----
+<1> <<entrypoints-search-session,Retrieve the `SearchSession`>>.
+<2> Start building the query as usual.
+<3> Mention that the expected result of the query is the highlights on the "title" field.
+An exception will be thrown if the field is not a full-text field with enabled highlighting or if it does not exist.
+<4> The predicate from the where clause will be used to determine which documents to include in the results
+and highlight the text from the "title" field.
+<5> Fetch the results, which will have highlighted fragments.
+====
+
+[NOTE]
+====
+Note that even though the "title" filed in the <<search-dsl-highlighting-using-highlight-projection,example above>>
+is a single value field, the resulting projection is a list of strings.
+This applies to any highlighting projection.
+====
+
+Highlight projections, same as <<search-dsl-projection-field,field projections>> can also be used in a combination with
+other projection types as well as with other highlight projections:
+
+[[search-dsl-highlighting-using-highlight-projection-composite]]
+.Using a composite highlight projection
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/highlighting/HighlightingDslIT.java[tags=composite]
+----
+<1> Start building the query as usual.
+<2> Add a regular field projection on a "title" field.
+<3> Add a highlight projection on a "description" field.
+<4> Note that to get a nonempty list for a highlight projection,
+the field we apply such projection to should be a part of a predicate.
+If there's no match within the field, we want to highlight - an empty list will be returned by default.
+See <<search-dsl-highlighting-highlighter-options-no-match-size,no match size configuration>> option for more details on how this can be adjusted.
+<5> Fetch the results.
+====
+
+Sometimes we might want to apply different highlighter types to various fields or specify some non-default configurations for a highlighter.
+To accomplish this, a highlight projection can have an optional `highlighter(..)` parameter set to a name of a named highlighter.
+As for the highlighter configurations - those can be passed to a query after the where clause.
+There are named highlighters and a query highlighter. Named highlighters must be of the same type as a query one if such was configured.
```

Comment:
yeah ... I wanted to have some synonyms so that default is not overused... but on the other hand 😃 as you've been saying it's a ref doc, so better be strict and use the "official names" 😄